File: src/main/java/chapter4/create/IntervalRangeExample.java
package chapter4.create;

import common.CommonUtils;
import common.Log;
import io.reactivex.Observable;

import java.util.concurrent.TimeUnit;


public class IntervalRangeExample {

    public void printNumbers() {
        Observable<Long> source = Observable.intervalRange(1, 5, 100L, 100L, TimeUnit.MILLISECONDS);
        source.subscribe(Log::it);
        CommonUtils.sleep(1000);
    }

    public void makeWithInterval() {
        Observable<Long> source = Observable.interval(100L, TimeUnit.MILLISECONDS)
                .map(val -> val + 1)
                .take(5);
        source.subscribe(Log::it);
        CommonUtils.sleep(1000);
    }

    public static void main(String[] args) {
        IntervalRangeExample demo = new IntervalRangeExample();
        demo.printNumbers();
        demo.makeWithInterval();
    }

}

File: src/main/kotlin/chapter4/IntervalRangeExample_.kt
package chapter4

import common.CommonUtils_
import common.Log_
import io.reactivex.Observable
import java.util.concurrent.TimeUnit


class IntervalRangeExample_ {

    fun printNumbers() {
        val source = Observable.intervalRange(1, 5, 100L, 100L, TimeUnit.MILLISECONDS)
        source.subscribe { Log_.it(it) }
        CommonUtils_.sleep(1000)
    }

    fun makeWithInterval() {
        val source = Observable.interval(100L, TimeUnit.MILLISECONDS)
                .map { it + 1 }
                .take(5)
        source.subscribe { Log_.it(it) }
        CommonUtils_.sleep(1000)
    }

}


fun main(args: Array<String>) {
    val demo = IntervalRangeExample_();
    demo.printNumbers()
    demo.makeWithInterval()
}
